src/main/kotlin/ai/digital/integration/server/deploy/internals/cluster/operator/AwsEksHelper.kt

```diff
@@ -1,24 +1,391 @@
+                "$stackStatusCmd",
+                logOutput = true,
+                throwErrorOnFailure = false)
+        if (result == "CREATE_COMPLETE") {
+            return true
+        } else if (result.contains("does not exist")) {
+            return false
+        }
+        return false
+    }
+
+
+    private fun verifyClusterStatus(totalTimeInSec: Int, sleepTime: Long, stackId: String, status: String) {
+        val awsEksProvider: AwsEksProvider = getProvider()
+        stackStatus(awsEksProvider, totalTimeInSec, sleepTime, stackId, status)
+    }
+
+    private fun stackStatus(awsEksProvider: AwsEksProvider, totalTimeInSec: Int, sleepTime: Long, stackId: String, status: String) {
+        val stackStatusCmd = "aws --region ${awsEksProvider.region.get()} " +
+                "cloudformation  list-stacks " +
+                "--query 'StackSummaries[?StackId==`$stackId`].StackStatus' " +
+                "--output text"
+        val stackStatus = wait(status,
+                stackStatusCmd,
+                "Stack",
+                totalTimeInSec,
+                sleepTime)
+        if (!stackStatus) {
+            throw RuntimeException("Resource STACK status is not $status")
+        }
+    }
+
+    private fun wait(status: String, command: String, resource: String, totalTimeInSec: Int = 3000, sleepTime: Long = 1000): Boolean {
+        val expectedEndTime = System.currentTimeMillis() + totalTimeInSec // 20 mins
+        while (expectedEndTime > System.currentTimeMillis()) {
+            val result = ProcessUtil.executeCommand(project,
+                    "$command",
+                    logOutput = true,
+                    throwErrorOnFailure = false)
+            if (result.contains("$status")) {
+                return true
+            } else if (result.contains("does not exist")) {
+                return false
+            }
+            project.logger.lifecycle("$resource resource  \"$status\" status not met, retry. ")
+            Thread.sleep(sleepTime) //5 mins
+        }
+        return false
+    }
+
+    private fun createStorageClass(efsStorageClassName: String) {
+        createStorageClassEFS(efsStorageClassName)
+        getKubectlHelper().setDefaultStorageClass("gp2", efsStorageClassName)
+    }
+
+    private fun createStorageClassEFS(storageClassName: String) {
+        if (!getKubectlHelper().hasStorageClass(storageClassName)) {
+            project.logger.lifecycle("Create storage class: {}", storageClassName)
+            val fileSystemId = getFileSystemId()
+            helmInstallAwsEfs(fileSystemId, storageClassName)
+        } else {
+            project.logger.lifecycle("Skipping creation of the existing storage class: {}", storageClassName)
+        }
+    }
+
+    private fun helmInstallAwsEfs(fileSystemId: String, storageClassName: String) {
+        ProcessUtil.executeCommand(project, "helm repo add stable https://charts.helm.sh/stable", logOutput = false, throwErrorOnFailure = false)
```

Comment:
That's not a good idea to use deprecated helm chart to create an EFS 
Check here: https://github.com/helm/charts/tree/master/stable/efs-provisioner

Instead it has to be created as described here: https://xebialabs.github.io/xl-deploy-kubernetes-operator/docs/manual/openshift#how-to-configure-aws-efs

For now what you can do is, to create EFS manually on our AWS instance and it will be reused in each test.
The part of EFS creation we will make configurable. 
Or to point to existing one, or to create it, with the approach i pointed above.

Reply:
done